tsdb/engine/tsm1/writer.go

```diff
@@ -264,9 +264,10 @@ func NewDirectIndex() TSMIndex {
 // directIndex is a simple in-memory index implementation for a TSM file.  The full index
 // must fit in memory.
 type directIndex struct {
-	mu     sync.RWMutex
-	size   uint32
-	blocks map[string]*indexEntries
+	mu               sync.RWMutex
+	size             uint32
+	blocks           map[string]*indexEntries
+	cachedSortedKeys []string
```

Comment:
sure

On Tue, Apr 19, 2016 at 7:59 AM Jason Wilder <notifications@github.com>
wrote:

> In tsdb/engine/tsm1/writer.go
> <https://github.com/influxdata/influxdb/pull/5582#discussion_r60176562>:
>
> > @@ -264,9 +264,10 @@ func NewDirectIndex() TSMIndex {
> >  // directIndex is a simple in-memory index implementation for a TSM file.  The full index
> >  // must fit in memory.
> >  type directIndex struct {
> > -	mu     sync.RWMutex
> > -	size   uint32
> > -	blocks map[string]*indexEntries
> > +	mu               sync.RWMutex
> > +	size             uint32
> > +	blocks           map[string]*indexEntries
> > +	cachedSortedKeys []string
>
> This will double the size of the index in memory when writing TSM files
> during compactions. Can you revert this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdata/influxdb/pull/5582/files/ff72a3debddbae9be9a92d59f66cc3bc6394e66c#r60176562>
>